src/NemoStuff.jl

```diff
@@ -0,0 +1,804 @@
+Base.:(:)(x::Int, y::Nothing) = 1:0
+
+export neg!
+
+function neg!(w::Vector{Int})
+    w .*= -1
+end
+
+sub!(z::T, x::T, y::T) where {T} = x - y
+
+sub!(z::Rational{Int}, x::Rational{Int}, y::Int) = x - y
+
+neg!(z::Rational{Int}, x::Rational{Int}) = -x
+
+add!(z::Rational{Int}, x::Rational{Int}, y::Int) = x + y
+
+mul!(z::Rational{Int}, x::Rational{Int}, y::Int) = x * y
+
+is_negative(x::Rational) = x.num < 0
+
+is_negative(n::Integer) = cmp(n, 0) < 0
+is_positive(n::Integer) = cmp(n, 0) > 0
```

Comment:
For a follow-up PR:
```suggestion
is_negative(n::T) where T<:Real = n < zero(T)
is_positive(n::T) where T<:Real = n > zero(T)
```
Of course that ignores some icky questions, like: should `is_negative(-0.0)` be `true` (matching `signbit(-0.0) == true`) or `false` (matching `-0.0 == 0.0` resp. `iszero(-0.0)` both holding)? But I think for our purposes the above makes most sense.